src/layers/CompletionLayerV2.ts

```diff
@@ -0,0 +1,195 @@
+import { Application, Point, Rectangle, RENDERER_TYPE, SimpleRope, Texture } from 'pixi.js';
+import { OnUpdateEvent, PixiRenderApplication } from '..';
+import { makeTubularPolygon } from '../datautils/wellboreItemShapeGenerator';
+import { Completion, foldCompletion, GenericScreen, GenericTubing, OnRescaleEvent } from '../interfaces';
+import { createNormals, offsetPoints } from '../utils/vectorUtils';
+import { WellboreBaseComponentLayer, WellComponentBaseOptions } from './WellboreBaseComponentLayer';
+
+interface TubularRenderingObject {
+  pathPoints: number[][];
+  polygon: Point[];
+  leftPath: Point[];
+  rightPath: Point[];
+  radius: number;
+}
+
+export interface CompletionLayerV2Options<T extends Completion[]> extends WellComponentBaseOptions<T> {
+  screenScalingFactor?: number;
+  screenLineColor?: number;
+  tubingScalingFactor?: number;
+}
+
+export class CompletionLayerV2<T extends Completion[]> extends WellboreBaseComponentLayer<T> {
```

Comment:
Use CompletionLayer as name. Old layer was never implemented and was just acting as a placeholder.